src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c

```diff
@@ -0,0 +1,911 @@
+		sprintf(msg, tmp, attr_name, attr_value, flags);
+		throw_exception(env, msg, rc);
+	}
+	(*env)->ReleaseStringUTFChars(env, name, attr_name);
+	(*env)->ReleaseStringUTFChars(env, value, attr_value);
+}
+
+JNIEXPORT jstring JNICALL Java_com_intel_daos_client_DaosFsClient_dfsGetExtAttr
+  (JNIEnv *env, jobject client, jlong dfsPtr, jlong objId, jstring name, jint expectedValueLen){
+	dfs_t *dfs = *(dfs_t**)&dfsPtr;
+	dfs_obj_t *file = *(dfs_obj_t**)&objId;
+	const char *attr_name = (*env)->GetStringUTFChars(env, name, NULL);
+	long value_len = expectedValueLen;
+	char *value = (char *)malloc(value_len+1); // 1 for \0
+	jstring ret = NULL;
+	if (!value) {
+		char *tmp = "Failed to allocate %d bytes for reading extended attribute value";
+		char *msg = (char *)malloc(strlen(tmp) + 4);
+		sprintf(msg, tmp, value_len);
+		throw_exception(env, msg, CUSTOM_ERR3);
+		goto out;
+	}
+	int rc = dfs_getxattr(dfs, file, attr_name, value, &value_len);
+	if (rc) {
+		char *tmp = "Failed to get ext attribute name: %s";
+		char *msg = (char *)malloc(strlen(tmp) + strlen(attr_name));
+		sprintf(msg, tmp, attr_name);
+		throw_exception(env, msg, rc);
+		goto out;
+	}
+	value[value_len] = '\0';
+	ret = (*env)->NewStringUTF(env, value);
+
+out:
+	(*env)->ReleaseStringUTFChars(env, name, attr_name);
+	if (value) free(value);
+	return ret;
+}
+
+JNIEXPORT void JNICALL Java_com_intel_daos_client_DaosFsClient_dfsRemoveExtAttr
+  (JNIEnv *env, jobject client, jlong dfsPtr, jlong objId, jstring name){
+	dfs_t *dfs = *(dfs_t**)&dfsPtr;
+	dfs_obj_t *file = *(dfs_obj_t**)&objId;
+	const char *attr_name = (*env)->GetStringUTFChars(env, name, NULL);
+	int rc = dfs_removexattr(dfs, file, attr_name);
+	if (rc) {
+		char *tmp = "Failed to remove ext attribute name: %s";
+		char *msg = (char *)malloc(strlen(tmp) + strlen(attr_name));
+		sprintf(msg, tmp, attr_name);
+		throw_exception(env, msg, rc);
+	}
+	(*env)->ReleaseStringUTFChars(env, name, attr_name);
+}
+
+JNIEXPORT jlong JNICALL Java_com_intel_daos_client_DaosFsClient_dfsGetChunkSize
+  (JNIEnv *env, jclass clientClass, jlong objId){
+	dfs_obj_t *file = *(dfs_obj_t**)&objId;
+	daos_size_t size;
+	int rc = dfs_get_chunk_size(file, &size);
+	if (rc) {
+		char *msg = "Failed to get chunk size of object. It's a directory, not a file? ";
+		throw_exception_const_msg(env, msg, rc);
+	}
+	return size;
+}
+
+JNIEXPORT jint JNICALL Java_com_intel_daos_client_DaosFsClient_dfsGetMode
+  (JNIEnv *env, jclass clientClass, jlong objId){
+	dfs_obj_t *file = *(dfs_obj_t**)&objId;
+	mode_t mode;
+	int rc = dfs_get_mode(file, &mode);
+	if (rc) {
+		char *msg = "Failed to get mode object";
+		throw_exception_const_msg(env, msg, rc);
+	}
+	return mode;
+}
+
+JNIEXPORT jboolean JNICALL Java_com_intel_daos_client_DaosFsClient_dfsIsDirectory
+  (JNIEnv *env, jclass clientClass, jint mode){
+	return S_ISDIR(mode) ? 1 : 0;
+}
+
+void JNI_OnUnload(JavaVM* vm, void *reserved) {
```

Comment:
(style)  "foo* bar" should be "foo *bar"
(style)  open brace '{' following function declarations go on the next line